team-last-donation/team-last-donation.js

```diff
@@ -1,75 +1,95 @@
-(function( $, ELT ){
+(function ($, ELT) {
     /**********
      * Main Functionality
      **********/
 
-	var teamMembers = {};
-	var newDonations = [];
-	var donationsSeen = 0;
-	var itemsToAdd = 1;
+    var $participants = {};
+    var $newDonations = [];
+    var donationsSeen = 0;
+    var itemsToAdd = 1;
+	// Structure placeholders
+    const $donation = $('#donation');
+    const $participantName = $('#participant-name');
+    const $trackingContainer = $('#tracking-container');
+    const $header = $('#header');
+    const $donationConjunction = $('#donation-conjunction');
 
     /* Initial setup of the layout and theme based on user settings */
     function start() {
-		const header = $('#header');
-
-		if (ELT.settings.showHeader) {
-			header.html(ELT.settings.headerMessage);
-		} else {
-			header.hide();
-		}
-
-		$('#donation-conjunction').html(ELT.settings.conjunctionText);
-
-		ELT.api.teamParticipants(ELT.settings.teamId, function(results) {
-			results.forEach(function(teamMember) {
-				teamMembers[teamMember.participantID] = teamMember;
-			});
-			
-			checkForNewDonations();
-			updateDonation();
-			setInterval(checkForNewDonations, ELT.settings.refreshTimeMS);
-			setInterval(updateDonation, ELT.settings.donationCycleMS); 
-		});   
-	}
+
+        if (ELT.settings.showHeader) {
+            $header.html(ELT.settings.headerMessage);
+        } else {
+            $header.remove();
+        }
+
+        if (ELT.settings.animate) {
+            $trackingContainer.addClass('animate')
+            // add animation direction
+            .addClass(`animate-${ELT.settings.animateTo}`);
+        }
+
+        if (ELT.settings.showRecipient) {
+            $donationConjunction.html(ELT.settings.conjunctionText);
+        } else {
+            $donationConjunction.remove();
+            $participantName.remove();
+        }
+
+        ELT.api.teamParticipants(ELT.settings.teamId, function (results) {
+            results.forEach(function (teamMember) {
+                $participants[teamMember.participantID] = teamMember;
+            });
+        });
+
+        checkForNewDonations();
+        updateDonation();
+        setInterval(checkForNewDonations, ELT.settings.refreshTimeMS);
+        setInterval(updateDonation, ELT.settings.donationCycleMS);
+    }
 
     /* Main loop */
     function checkForNewDonations() {
-		ELT.api.teamDonations(ELT.settings.teamId, checkForNewDonationsOnSuccess);
+        ELT.api.teamDonations(ELT.settings.teamId, checkForNewDonationsOnSuccess);
+    }
+
+    function checkForNewDonationsOnSuccess(results) {
+        if (results.length > donationsSeen) {
```

Comment:
I was able to figure out the query parameters for filtering by date:

`?where=createdDateUTC > 2023-10-04T15:22:59.267&orderBy=createdDateUTC ASC`

Would be nice to be able to add `donationsSince` to the config and default it to `Date.now()`.  Especially for the Team widget.  I didn't have time to work that one out tonight.